exercises/queen-attack/example.exs

```diff
@@ -38,6 +39,15 @@ defmodule Queens do
     white_x == black_x || white_y == black_y || diagonal?(white, black)
   end
 
+  defp is_valid_position?({x, y}) do
+    x in @board_range && y in @board_range
+  end
```

Comment:
This should either be made into a guard function:

```elixir
defguardp is_coodinate(c)
  when is_tuple(c)
  and tuple_size(c) == 2
  and c |> elem(0) |> is_integer()
  and c |> elem(1) |> is_integer()

defguardp is_board_position(p)
  when is_coordinate(p)
  and p |> elem(0) |> Kernel.in(@board_range)
  and p |> elem(1) |> Kernel.in(@board_range)
```

or be renamed without the `is_` prefix.  I am inclined towards the former just because I think guards are under expressed in the exercises.

Reply:
and just because it is defined as a guard doesn't mean we can't use it as a regular function, it just does additional compiler checks and ensures that it doesn't have any side effects.